pkg/availability/multisigcollector/parts/batchreconstruction/batchreconstruction.go

```diff
@@ -55,39 +53,32 @@ func IncludeBatchReconstruction(
 		//TODO this switch might be removable with some refactoring
 		// NOTE: it is assumed that cert is valid.
 
-		reqID := state.NextReqID
-		state.NextReqID++
-		state.RequestState[reqID] = &RequestState{
-			NonEmptyCerts: make([]msctypes.BatchIDString, 0),
-			Txs:           make(map[msctypes.BatchIDString][]*requestpbtypes.Request),
-			ReqOrigin:     origin,
-		}
-
 		mscCertsWrapper, ok := cert.Type.(*apbtypes.Cert_Mscs)
 		if !ok {
 			return fmt.Errorf("unexpected certificate type")
 		}
-		certs := mscCertsWrapper.Mscs
+		certs := mscCertsWrapper.Mscs.Certs
 
-		nonEmptyCerts := make([]*mscpbtypes.Cert, 0)
-		for _, c := range certs.Certs {
-			if !emptycert.IsEmpty(c) {
-				nonEmptyCerts = append(nonEmptyCerts, c) // do not add to requests
-			}
+		// An empty certificate (i.e., one certifying the availability of no transactions)
+		// corresponds to an empty list of transactions. Respond immediately.
+		if len(certs) == 0 {
+			apbdsl.ProvideTransactions(m, origin.Module, []*requestpbtypes.Request{}, origin)
+			return nil
 		}
 
-		if len(nonEmptyCerts) > 0 {
-			for _, c := range nonEmptyCerts {
-				batchIDString := msctypes.BatchIDString(c.BatchId)
-				state.RequestState[reqID].NonEmptyCerts = append(state.RequestState[reqID].NonEmptyCerts, batchIDString) // save the order in which the batches were decided
-				state.RequestState[reqID].Txs[batchIDString] = nil
-				batchdbpbdsl.LookupBatch(m, mc.BatchDB, c.BatchId, &lookupBatchLocallyContext{c, origin, reqID})
+		reqID := state.NextReqID
+		state.NextReqID++
+		state.RequestState[reqID] = &RequestState{
+			NonEmptyCerts: make([]msctypes.BatchIDString, 0),
+			Txs:           make(map[msctypes.BatchIDString][]*requestpbtypes.Request),
+			ReqOrigin:     origin,
+		}
 
-			}
-		} else {
-			// give empty batch and delete request
-			apbdsl.ProvideTransactions(m, origin.Module, []*requestpbtypes.Request{}, origin)
-			delete(state.RequestState, reqID)
+		for _, c := range certs {
+			batchIDString := msctypes.BatchIDString(c.BatchId)
```

Comment:
I think my above comment may be deprecated after seeing #417 , but leaving it for completeness and for you to double check this important bit.

Reply:
Yes, if `certs[i]` is nil, this would crash, but `certs[i]` is never nil, even irrespectively of #417. Already in this version (in fact, even before this PR), correct code never produces a nil entry in this list (not even ISS or the batch fetcher). What could happen, though, that a malicious node proposes a malformed certificate with a `nil` entry (once `nil` entries are supported - until then, we would crash even before getting here). This is a matter of input sanitization though, there already is an issue for that.